sorting/tree_sort.cpp

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file
+ * @brief Tree Sort Algorithm 
+ * More about it: https://en.wikipedia.org/wiki/Tree_sort
```

Comment:
Markdown link formatting should be used here instead.
```suggestion
 * @brief Implementation of a [tree sorting](https://en.wikipedia.org/wiki/Tree_sort) algorithm
```